contributors/design-proposals/propagation.md

```diff
@@ -109,6 +125,12 @@ Opinion against this:
 
 1. (From @jonboulle) May cause cross-runtime compatibility issue.
 
+1. It's not possible to validate a pod + mount propagation. Mount propagation
```

Comment:
@thockin I've been toying around with the idea of contributing a 'namedEmptyDir' volume type. what it does, is similar to an emptyDir, but it is given a name unique to a namespace, and its lifecycle doesn't have it deleted until the last pod on the same host that references it goes away.
So, it would be stored somwhere like /var/lib/kubelet/volumes/namedEmptydir/<namespace name>/<name from user>/ (or whatever)

This would allow multiple k8s objects such as deamonsets or daemonsets/deployments/etc to coexist on the host and share some transient storage amongst them but without having to book keep where to store them using a hostpath and ensuring different name spaces don't stop on each others data.

Its use works in concert with this proposal in that say you have a fuse based system such as cvmfs that has a backing cache. You could have several different cvmfs daemonsets or pods consume the cache dir from a namedEmptyDir on the host via "cvmfs-cache" and have another one called "cvmfs" with mount propagation set to shared where cvmfs automounter is running.  Any pod in the namespace can then mount the namedEmptyDir 'cvmfs' and the the automount cvmfs volume would work optimally.

Ideally for security, you would have the volume mounted as shared on the autofs fuse filesystem container and have it slave on the things consuming it. Even more reason for having it as a flag and not a volumetype.